### PR TITLE
Add NTIA minimum SBOM elements

### PR DESCRIPTION
**Pull request by j-hartley**
Problem:

The BOM generated by syft does not contain all the recommended minimum
data for SBOM best practice, see:
https://www.ntia.doc.gov/report/2021/minimum-elements-software-bill-materials-sbom

Solution:

Patch the missing elements into the SBOM prior to uploading.
Validate the SBOM against the cyclone DX 1.2 XML schema to make sure it
is correctly constructed.

### CODE REVIEW

File: scripts/sbom_scraper.sh [hunk 1]
```diff
@@ -33,19 +33,38 @@ then
...
+SUPPLIER_URL=https://hub.docker.com
+TOOL_NAME="$(git config --get remote.origin.url) $(git ls-files --full-name "$SCRIPTNAME")"
+TOOL_VERSION=$(git describe --tags)${GIT_STATUS:++}
+TOOL_VENDOR="Jitsuin Inc"
+TOOL_HASH_ALG=SHA-256
+# shellcheck disable=SC2002
+TOOL_HASH_CONTENT=$(cat "$0" | openssl dgst -sha256)
 
-# credentials directory has 0600 permissions
+# credentials directory should have 0600 permissions
```
Comment (eccles) [inline scripts/sbom_scraper.sh:67]: actually it is 0700

File: scripts/sbom_scraper.sh [hunk 2]
```diff
@@ -55,37 +74,44 @@ URL=https://app.rkvst.io
 usage() {
     cat >&2 <<EOF
 
-Scrape an SBOM from a docker image and upload to abom archivist
+Create a Cyclone DX 1.2 XML SBOM from a docker image and upload to RKVST SBOM Hub
 
-Usage: $SCRIPTNAME [-p] [-c clientsecretfile] [-o output format] [-s sbomFile ] [-u url] client_id [docker-image|sbom file]
+Usage: $SCRIPTNAME [-a AUTHOR_NAME] [-A AUTHOR_NAME] [-c CLIENT_SECRET_FILE] [-e AUTHOR_EMAIL] [-s] [-p] [-u URL] CLIENT_ID [docker-image:tag|sbom file]
```
Comment (eccles) [inline scripts/sbom_scraper.sh:79]: AUTHOR_NAME twice? COMPONENT_AUTHOR_NAME perhaps for -A

File: scripts/sbom_scraper.sh [hunk 3]
```diff
@@ -133,19 +159,179 @@ SECRET=$(cat "${CLIENTSECRET_FILE}")
...
     OUTPUT="${TEMPDIR}/${OUTFILE}"
     ${SYFT} -q packages -o "${FORMAT}" "${DOCKER_IMAGE}"> "${OUTPUT}"
 else
     OUTPUT="${DOCKER_IMAGE}"
 fi
 
+# ----------------------------------------------------------------------------
+# Update SBOM including NTIA minimum elments
+# ----------------------------------------------------------------------------
+ORIG_COMPONENT_NAME=$(xq -r .bom.metadata.component.name "$OUTPUT")
```
Comment (eccles) [inline scripts/sbom_scraper.sh:172]: whwre does xq come from? its not checked at top of file?

File: scripts/sbom_scraper.sh [hunk 4]
```diff
@@ -133,19 +159,179 @@ SECRET=$(cat "${CLIENTSECRET_FILE}")
...
+[ -z "$SUPPLIER_URL" ] && echo >&2 "Unable to determine component supplier url" && exit 1
+[ -z "$COMPONENT_AUTHOR_NAME" ] && echo >&2 "Unable to determine component author name" && exit 1
+[ -z "$COMPONENT_NAME" ] && echo >&2 "Unable to determine component name" && exit 1
+[ -z "$COMPONENT_VERSION" ] && echo >&2 "Unable to determine component version" && exit 1
+[ -z "$COMPONENT_HASH_ALG" ] && echo >&2 "Unable to determine component hash algorithm" && exit 1
+[ -z "$COMPONENT_HASH_CONTENT" ] && echo >&2 "Unable to determine component hash content" && exit 1
+
+PATCHED_OUTPUT="${OUTPUT}.patched"
+
+python3 <(cat <<END
```
Comment (eccles) [inline scripts/sbom_scraper.sh:227]: maybe add a check for python3 at top of file ?

**Review by eccles — CHANGES_REQUESTED** [review]

File: scripts/sbom_scraper.sh (hunk 4 above)
Comment (j-hartley) [reply scripts/sbom_scraper.sh:216]: done, thanks

File: scripts/sbom_scraper.sh (hunk 3 above)
Comment (j-hartley) [reply scripts/sbom_scraper.sh:161]: added check, thanks

File: scripts/sbom_scraper.sh (hunk 2 above)
Comment (j-hartley) [reply scripts/sbom_scraper.sh:68]: updated, thanks

File: scripts/sbom_scraper.sh (hunk 1 above)
Comment (j-hartley) [reply scripts/sbom_scraper.sh]: updated, thanks

**Review by eccles — APPROVED** [review]

**Pull request merged**